packages/docs-page/index.tsx

```diff
@@ -127,7 +144,26 @@ export function DocsPageWrapper({
   )
 }
 
-export default function DocsPage({
+export interface DocsPageProps {
+  product: { name: string; slug: string }
+  baseRoute: string
+  showEditPage: boolean
+  additionalComponents: MDXProviderComponentsProp
+  staticProps: {
+    mdxSource: MDXRemoteSerializeResult
+    frontMatter: {
+      canonical_url: string
+      description: string
+      page_title: string
+    }
+    currentPath: string
+    navData: NavData
+    githubFileUrl: string
+    versions: { name: string; label: string }[]
+  }
+}
+
+const DocsPage: ComponentType<DocsPageProps> = ({
```

Comment:
As `DocsPage` doesn't accept children, I'm not sure it makes sense to use `ComponentType` here, which defines props as `PropsWithChildren<DocsPageProps>`. I would maybe just type props here:
```ts
export default function DocsPage({ ... }: DocsPageProps) {
```

ref: https://sourcegraph.com/github.com/DefinitelyTyped/DefinitelyTyped/-/blob/types/react/index.d.ts?L82:10

Reply:
I'll opt for your suggestion, but will need to add an explicit return type `: JSX.Element`

We have a linting rule — `Missing return type on function. eslint@typescript-eslint/explicit-module-boundary-types` — that warns if you don't have an explicit return type.
- (Maybe we should re-assess this rule? Type-inference is pretty good, and sometimes explicit typing is prone to human error — 0 warnings, but incorrect type)

Reply:
Ah yeah, that rule! I think the value of that rule is debatable for react components, but it can be handy for other methods. We have had a few discussions about this rule in the past 💭  cc @hashicorp/web-platform

I think the return type should be `React.ReactElement`, FWIW. This matches the definition of the component types from `@types/react`.

Reply:
I was looking at `React.ReactElement` (from the Sourcegraph link) ... vs. `JSX.Element` (from type-hint/inference) 😅 

I'll update to `React.Element`!